cartography/intel/aws/permission_relationships.py

```diff
@@ -0,0 +1,338 @@
+                # The action is allowed by this policy
+                return True, False
+    # The action is not allowed by this policy, but not specifically denied either
+    return False, False
+
+
+def principal_allowed_on_resource(policies, resource_arn, permissions):
+    """ Evaluates an enture set of policies for a specific resource for a specific permission.
+
+
+    Arguments:
+        policies {[dict]} -- The policys to evaluate
+        resource_arn {str} -- The resource to test the permission against
+        permissions {[str]} -- The permissions to evaluate
+
+    Returns:
+        [bool] -- True if the policies allow the permission against the resource
+    """
+    granted = False
+    for _, statements in policies.items():
+        allowed, explicit_deny = evaluate_policy_for_permission(statements, permissions, resource_arn)
+
+        if explicit_deny:
+
+            return False
+        if not granted and allowed:
+            granted = True
+
+    return granted
+
+
+def calculate_permission_relationships(principals, resource_arns, permission):
+    """ Evaluate principals permissions to resources
+    This currently only evaluates policies on IAM principals. It does not take into account
+    Resource Policies - Policies attached to the resource instead of the IAM principal
+    Permission Boundaries - Boundaries for an IAM principal
+    Session Policies - Special policies for Federated users
+
+    AWS Policy evaluation reference
+    https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_evaluation-logic.html
+
+    Arguments:
+        principals {[dict]} -- The principals to check permission for
+        resource_arns {[str]} -- The resources to test the permission against
+        permissions {[str]} -- The permissions to evaluate
+
+    Returns:
+        [dict] -- The allowed mappings
+    """
+    allowed_mappings = []
+    for resource_arn in resource_arns:
+        for principal_arn, policies in principals.items():
+            if principal_allowed_on_resource(policies, resource_arn, permission):
+                allowed_mappings.append({"principal_arn": principal_arn, "resource_arn": resource_arn})
+    return allowed_mappings
+
+
+def parse_statement_node(node_group):
+    """ Parse a dict from group of Neo4J node
+
+    Arguments:
+        node_group {[Neo4j.Node]} -- the node to parse
+
+    Returns:
+        [list] -- A list of statements from the node
+    """
+    return [n._properties for n in node_group]
+
+
+def compile_regex(item):
+    """ Compile a clause into a regex. Clause checking in AWS is case insensitive
+
+    Arguments:
+        item {str} -- the item to create the regex for
+
+    Returns:
+        [re.Pattern] -- The precompiled regex pattern.
+    """
+    if isinstance(item, str):
+        item = item.replace(".", "\\.").replace("*", ".*")
+        item = re.compile(item, flags=re.IGNORECASE)
```

Comment:
As we talked about, we should surround this with `try-except` for `sre_constants.error`. 

Something like

```python
try:
    re.compile( ... )
except sreconstants.error as e:
    logger.warning("Skipping clause '%s'; could not parse regex due to error '%s'", item, e)
```

Reply:
fixed